### ISSUE TITLE
Silent success when languageOptions.sourceType is incorrectly defined

### ISSUE THREAD

**Issue by DamienCassou** [eslint-community/eslint-plugin-eslint-plugin#483]
I'm migrating my eslint plugin to v9 and forgot to set `languageOptions.sourceType` to "commonjs" so the default value was "module". Running eslint on my rules would produce 0 warnings or errors because `getRuleExportsESM()` (from `util.js`) returns an empty object so `getRuleInfo()` returns `null` and `no-missing-message-ids.js` (for example) has this code:

```js
const ruleInfo = utils.getRuleInfo(sourceCode);
if (!ruleInfo) {
  return {};
}
```

Would it be possible to get warnings/errors in such cases?

**Comment by aladdin-add** [comment]
> forgot to set languageOptions.sourceType to "commonjs" so the default value was "module".

in such cases, the rule [`no-undef`](https://eslint.org/docs/latest/rules/no-undef) could report an error when using `require/module` - they can only be used in cjs.

**Comment by DamienCassou** [comment]
Thank you :+1:
